### ISSUE TITLE
Exception when calling `Plan.Get(string)`

### ISSUE THREAD

**Issue by CodingGorilla** [recurly/recurly-client-net#205]
I have started getting the following exception after upgrading to the latest recurly-net-client package, and it seems related to this commit: https://github.com/recurly/recurly-client-net/commit/881db5ca1ea75b93368235e08512245a6e7961e8

Here's the stack trace:

```
   at System.Number.StringToNumber(String str, NumberStyles options, NumberBuffer& number, NumberFormatInfo info, Boolean parseDecimal)
   at System.Number.ParseInt32(String s, NumberStyles style, NumberFormatInfo info)
   at System.Xml.XmlReader.ReadElementContentAsInt()
   at Recurly.Plan.ReadXml(XmlTextReader reader) in C:\Users\ben\Documents\GitHubVisualStudio\recurly-client-net\Library\Plan.cs:line 300
   at Recurly.Client.ReadWebResponse(HttpWebResponse response, ReadXmlDelegate readXmlDelegate, ReadXmlListDelegate readXmlListDelegate, ReadResponseDelegate responseDelegate) in C:\Users\ben\Documents\GitHubVisualStudio\recurly-client-net\Library\Client.cs:line 381
   at Recurly.Client.PerformRequest(HttpRequestMethod method, String urlPath, WriteXmlDelegate writeXmlDelegate, ReadXmlDelegate readXmlDelegate, ReadXmlListDelegate readXmlListDelegate, ReadResponseDelegate reseponseDelegate) in C:\Users\ben\Documents\GitHubVisualStudio\recurly-client-net\Library\Client.cs:line 168
   at Recurly.Plans.Get(String planCode) in C:\Users\ben\Documents\GitHubVisualStudio\recurly-client-net\Library\Plan.cs:line 403
   at Recurly.Subscription.get_Plan() in C:\Users\ben\Documents\GitHubVisualStudio\recurly-client-net\Library\Subscription.cs:line 76
   at DataMateWeb.Services.Subscriptions.RecurlyProvider.Actions.GetPlanFromSubscriptionAction.<>c__DisplayClass0_0.<Execute>b__0() in C:\Users\codin\Source\Repos\DataMateWeb.Services.Subscriptions\src\DataMateWeb.Services.Subscriptions.RecurlyProvider\Actions\GetPlanFromSubscriptionAction.cs:line 18
   at DataMateWeb.Services.Subscriptions.RecurlyProvider.Actions.RecurlyActionBase.TryRecurlyAction[T](Func`1 recurlyAction) in C:\Users\codin\Source\Repos\DataMateWeb.Services.Subscriptions\src\DataMateWeb.Services.Subscriptions.RecurlyProvider\Actions\RecurlyActionBase.cs:line 55
```

It seems to be that it's attempting to read the `total_billing_cycles` element, which in my case looks like:
```
<total_billing_cycles nil="nil"></total_billing_cycles>
```

I'm not overly familiar with the recurly code, but it looks like it's missing a null (or nill) check here, but I'm not 100% sure how to add it or I would submit a pull request to fix it.

**Comment by 1TheMuffinMan** [comment]
Thanks a lot for including descriptive information in this issue as well as the stack trace. It helps a lot with tracking down issues fast and getting fixes out quickly. 👍 

It looks like the offending line in `Plan.cs` is here https://github.com/recurly/recurly-client-net/blob/master/Library/Plan.cs#L300

**Comment by CodingGorilla** [comment]
@1TheMuffinMan Yea, this seems to be an issue in the `XmlReader` itself.  If you look at: https://referencesource.microsoft.com/#System.Xml/System/Xml/Core/XmlReader.cs,295 (and follow the calls into `XmlConvert.ToInt32`) it does no null or empty checking at all on that value.  So I think you'd have to read this as a string and do the conversion yourself.

**Comment by 1TheMuffinMan** [comment]
@CodingGorilla Thanks for confirming via the MS source code. I believe we do a similar thing with `DateTime` (read as string and then do a TryParse). I might be able to fix this tonight and add a unit test.

**Comment by SuperStromboli** [comment]
For what it's worth, I was able to repro this.

**Comment by bhelx** [comment]
@CodingGorilla Thanks for catching so quick. This was something that was changed in the latest 1.4.6 release. I'm going to have a fix ready in a moment and I'll get a release out.

**Comment by bhelx** [comment]
I've added a fix for this in #206 and I've pushed this to nuget as 1.4.7:

Here is the hotfix release:
https://github.com/recurly/recurly-client-net/releases/tag/1.4.7

Thanks @CodingGorilla and @SuperStromboli for pointing it out.

On another note, I didn't write a specific test for it because I think we should refactor the parsers to have a generic way to return nulls and then add tests to that. We've made this mistake more than once and I think we should solve it generically. This will be on my list for Q1 this year.
